Realm/Realm.Weaver/Analytics/Metric.cs

```diff
@@ -55,83 +66,132 @@ public static class Framework
-            ["ProjectId"] = "Anonymized Bundle ID",
-            ["RealmSdk"] = "Binding",
-            ["Language"] = "Language",
-            ["LanguageVersion"] = "Language Version",
-            ["HostOsType"] = "Host OS Type",
-            ["HostOsVersion"] = "Host OS Version",
-            ["HostCpuArch"] = "Host CPU Arch",
-            ["TargetOsType"] = "Target OS Type",
-            ["TargetOsMinimumVersion"] = "Target OS Minimum Version",
-            ["TargetOsVersion"] = "Target OS Version",
-            ["TargetCpuArch"] = "Target CPU Arch",
-            ["RealmSdkVersion"] = "Realm Version",
-            ["CoreVersion"] = "Core Version",
-            ["IsSyncEnabled"] = "Sync Enabled",
-            ["FrameworkUsedInConjunction"] = "Framework", // this refers to UI frameworks and similar Realm is used together with
-            ["FrameworkUsedInConjunctionVersion"] = "Framework Version",
-            ["SdkInstallationMethod"] = "Installation Method",
-            ["IdeUsed"] = "IDE",
-            ["IdeUsedVersion"] = "IDE Version", // this holds info about the msbuild version
-            ["NetFramework"] = "Net Framework",
-            ["NetFrameworkVersion"] = "Net Framework Version"
-        };
+            public const string UserId = "distinct_id";
+            public const string ProjectId = "Anonymized Bundle ID";
+            public const string RealmSdk = "Binding";
+            public const string Language = "Language";
+            public const string LanguageVersion = "Language Version";
+            public const string HostOsType = "Host OS Type";
+            public const string HostOsVersion = "Host OS Version";
+            public const string HostCpuArch = "Host CPU Arch";
+            public const string TargetOsType = "Target OS Type";
+            public const string TargetOsMinimumVersion = "Target OS Minimum Version";
+            public const string TargetOsVersion = "Target OS Version";
+            public const string TargetCpuArch = "Target CPU Arch";
+            public const string RealmSdkVersion = "Realm Version";
+            public const string CoreVersion = "Core Version";
+            public const string IsSyncEnabled = "Sync Enabled";
+            public const string FrameworkUsedInConjunction = "Framework"; // this refers to UI frameworks and similar Realm is used together with
+            public const string FrameworkUsedInConjunctionVersion = "Framework Version";
+            public const string SdkInstallationMethod = "Installation Method";
+            public const string IdeUsed = "IDE";
+            public const string IdeUsedVersion = "IDE Version"; // this holds info about the msbuild version
+            public const string NetFramework = "Net Framework";
+            public const string NetFrameworkVersion = "Net Framework Version";
+        }
+
+        public static class Feature
```

Comment:
After some deliberation, I decided it's a good idea to add the features as constants to make sure we don't introduce bugs via typos. We still have the dictionary and there are some features where we use the strings (e.g. ``List`1``).